### DISCUSSION TITLE
How do I make military seperated from regular civ vehicles?

### DISCUSSION THREAD

**Discussion by OfirAviv** [Ppgtjmad/SimpleShops#74] [Q&A] [answered]
as the question states, How do I make the military separated from regular civ vehicles?
I could use this option for sure, even making it so only some ranks can access certain vehicles is good for me, I'd love some help in this matter, thank you! :D

**Comment by Ppgtjmad** [comment, marked as answer]
Hello @OfirAviv,

I guess you could use this config ([HG_VehiclesShop.cfg](https://github.com/Ppgtjmad/SimpleShops/blob/master/HG/Config/HG_VehiclesShopCfg.h)):

```
class HG_CivilianShop
{
    conditionToAccess = "playerSide isEqualTo civilian";
	
    class LandVehicles
    {
        displayName = "Land Vehicles";
        vehicles[] =
        {
            {"C_SUV_01_F",15000,"true"}
        };
        spawnPoints[] =
        {
            {"$STR_HG_MARKER_1",{"civilian_vehicles_spawn_1"}}
        };
    };
};

class HG_MilitaryShop
{
    conditionToAccess = "playerSide isEqualTo west";

    class LandVehicles
    {
        displayName = "Land Vehicles";
        vehicles[] =
        {
            {"B_MRAP_01_F",45000,"true"}
        };
        spawnPoints[] =
        {
            {"$STR_HG_MARKER_2",{"military_vehicles_spawn_1"}}
        };
    };
};
```

Then the actions to open them:

**Civilian Vehicles Shop**
```
this addAction["<img image='HG\UI\Icons\money.paa' size='2' shadow='0'/><t color='#FF0000' font='RobotoCondensedBold'>Civilian Vehicles Shop</t>",{_this call HG_fnc_dialogOnLoadVehicles},"HG_CivilianShop",0,false,false,"",'(alive player) && !dialog && player distance _target < 3'];
```

**Military Vehicles Shop**
```
this addAction["<img image='HG\UI\Icons\money.paa' size='2' shadow='0'/><t color='#FF0000' font='RobotoCondensedBold'>Military Vehicles Shop</t>",{_this call HG_fnc_dialogOnLoadVehicles},"HG_MilitaryShop",0,false,false,"",'(alive player) && !dialog && player distance _target < 3'];
```

**Comment by OfirAviv** [comment]
hello! @Ppgtjmad :), thank you for your help, one last question,

so whenever I define a different class then I can "post them" on the init of the in-game object as that class within the file/function call?

in this case, calling either classes `HG_MilitaryShop` or `HG_CivilianShop` within the call of `HG_fnc_dialogOnLoadVehicles`  which calls to the vehicle shop config file?

**Comment by Ppgtjmad** [comment]
To make it simple:

Theses are classnames -> `HG_MilitaryShop` `HG_CivilianShop`
They are defined in `HG_VehiclesShopCfg` where you setup all your vehicles shops.
These classnames are used as parameters for the function call `HG_fnc_dialogOnLoadVehicles`.
They serve as a reference to what config class you want to pull data from.

As you can see here at the very top in [fn_dialogOnLoadVehicles.sqf](https://github.com/Ppgtjmad/SimpleShops/blob/master/HG/Functions/Client/VehiclesShop/fn_dialogOnLoadVehicles.sqf):

`params["_target","_caller","_id","_whatShop"];`

All of these params are passed by the `addAction` command and `_whatShop` is the one holding your shop config class.

**Reply by OfirAviv** [reply]
Got it, thank you! :)
